### PR TITLE
kubernetes-helm-mapkubeapis: init at 0.4.1

### PR DESCRIPTION
**Pull request by aos**
## Description of changes

Was looking for this plugin, but couldn't find it. I found this open PR:
https://github.com/NixOS/nixpkgs/pull/217530 but it seems to have gone
stale (no activity in over a year). I posted in it but figured everyone
is too busy so I just decided to resubmit, addressing comments :-)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by lolbinarycat — CHANGES_REQUESTED** [review]
looks pretty good, but there's a few issues:
1. interpolating pname (expecially into `repo`) is discouraged
2. `with lib` is a bit controversial, and can sometimes cause issues (in one case the version of a package was accidentally replaced with the version of nixpkgs itself).

**Comment by aos** [comment]
Good to know - thank you. I've addressed your comments. I looked around at other recently `init` packages, and they seem to go with `with lib.{licenses,maintainers}`, so I chose to do that as well.

File: pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  # NOTE: Remove the install and upgrade hooks.
+  postPatch = ''
+    sed -i '/^hooks:/,+2 d' plugin.yaml
+  '';
+
+  postInstall = ''
+    install -dm755 $out/${pname}
+    mv $out/bin $out/${pname}/
+    install -m644 -Dt $out/${pname}/config/ config/Map.yaml
+    install -m644 -Dt $out/${pname} plugin.yaml
```
Comment (nevivurn) [inline pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix:25]:
```suggestion
    install -dm755 $out/helm-mapkubeapis
    mv $out/bin $out/helm-mapkubeapis/
    install -m644 -Dt $out/helm-mapkubeapis/config/ config/Map.yaml
    install -m644 -Dt $out/helm-mapkubeapis plugin.yaml
```

Interpolating `pname` is discouraged.

**Review by nevivurn — APPROVED** [review]

**Review by lolbinarycat — APPROVED** [review]

File: pkgs/applications/networking/cluster/helm/plugins/default.nix [hunk 2]
```diff
@@ -14,4 +13,5 @@
 
   helm-unittest = callPackage ./helm-unittest.nix { };
 
+  helm-mapkubeapis = callPackage ./helm-mapkubeapis.nix { };
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/helm/plugins/default.nix:16]: Please keep the file alphabetically sorted

File: pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  postInstall = ''
+    install -dm755 $out/helm-mapkubeapis
+    mv $out/bin $out/helm-mapkubeapis/
+    install -m644 -Dt $out/helm-mapkubeapis/config/ config/Map.yaml
+    install -m644 -Dt $out/helm-mapkubeapis plugin.yaml
+  '';
+
+  meta = {
+    description =
+      "A Helm plugin which maps deprecated or removed Kubernetes APIs in a release to supported APIs";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix:30]:
```suggestion
    description = "Helm plugin which maps deprecated or removed Kubernetes APIs in a release to supported APIs";
```

Comment (aos) [reply pkgs/applications/networking/cluster/helm/plugins/helm-mapkubeapis.nix:30]: I ran it through `nixpkgs-fmt` and this is what it spit out, but I presume that's not standard :-) I'll commit your suggestion. Thank you

File: pkgs/applications/networking/cluster/helm/plugins/default.nix (hunk 2 above)
Comment (aos) [reply pkgs/applications/networking/cluster/helm/plugins/default.nix:16]: It's not already sorted though hence why I added it to the bottom. Let me sort the file itself.

**Comment by SuperSandro2000** [comment]
@ofborg build kubernetes-helmPlugins.helm-mapkubeapis

**Pull request merged**